### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by teekaytech**
**In this milestone, we added a new presentational component (CategoryFilter) which is just a simple select box.**

**In addition, we created a new Redux action (CHANGE_FILTER) and a filter.js reducer.**

### CODE REVIEW

File: src/components/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { connect } from 'react-redux';
+
+class CategoryFilter extends Component {
+  handleFilterChange = category => {
+    const { bookFilter } = this.props;
+    bookFilter(category);
+  }
+
+  render() {
+    const categories = [
```
Comment (mosaaleb) [inline src/components/CategoryFilter.js:12]: - [ ] All is not a valid book category.

File: src/reducers/filter.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const filter = (state = [], action) => {
```
Comment (mosaaleb) [inline src/reducers/filter.js]: - [x] It defaults to 'All'

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Changes Required

Well done @teekaytech @evabanegacom :clap:. Most requirements are handled, yet there are some we need to work on before moving on.

Please take a look at my comments below.

##### Muhammad Ebeid

File: src/components/CategoryFilter.js (hunk 1 above)
Comment (teekaytech) [reply src/components/CategoryFilter.js:12]: Thank you for your feedback.

The last part of the statement says **...although it should be a filter option**, and that is exactly what we did. We know that our `All` should not be saved in the Redux store and we never did. I urge the next reviewer to please check out the `filter` reducer and see how we ensured that the `All` option is not stored in our database.

Comment (bellom) [reply src/components/CategoryFilter.js:12]: ### [Optional]
According to the instruction, our CategoryFilter should be a presentational component;
image
therefore; our CategoryFilter should not be connected to the redux store.

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**